test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -81,33 +82,38 @@ def test_create_duplicate_cname_record_fails
   end
 
   def test_remove_address_records_with_longest_zone_match
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:get_ipv4_address!).with('host.foo.bar.domain.local')
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:enum_records).with('bar.domain.local', 'host.foo.bar.domain.local',  'A').returns(['1.1.1.1','2.2.2.2'])
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:remove_specific_record_from_zone).with('bar.domain.local', 'host.foo.bar.domain.local', '1.1.1.1', 'A').returns(true)
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:remove_specific_record_from_zone).with('bar.domain.local', 'host.foo.bar.domain.local', '2.2.2.2', 'A').returns(true)
     assert_nil @server.remove_a_record('host.foo.bar.domain.local')
```

Comment:
I think provider tests can be broken up (which will make them simpler too): the methods that's been pulled into dns_common (create_* and delete_*) can be tested in their own suite. Provider's implementations of do_create and do_delete methods can be tested in their own suites, which will reduce the amount of mocking and make these tests less fragile.